app/templates/main/upload.html

```diff
@@ -1,11 +1,24 @@
 {% extends "base.html" %}
 
+{%- from 'govuk_frontend_jinja/components/file-upload/macro.html' import govukFileUpload -%}
+
 {% block beforeContent %}
 {{ super() }}
 {% endblock beforeContent %}
 
 {% block content %}
 
-    [upload file here]
+    <form method="post" action="{{ url_for('main.upload') }}" enctype="multipart/form-data">
+        {{ govukFileUpload(fixtures[0]['options'])}}
+        <input type="submit" value="Upload" />
+    </form>
+
+    {% if response.validation_errors %}
+        {% for error in response.validation_errors %}
+            <p> {{ error }} </p>
+        {% endfor %}
+    {% else %}
+        <p>{{ response }}</p>
+    {% endif %}
```

Comment:
Don't think you need to return the response as a paragraph (especially as it looks like it'll be set as an empty string by default). 

I also don't think you necessarily need to set response as a particular value in the GET request, and that way you can just do {% if response %}.